packages/components/src/components/ScalarTooltip/useTooltip.ts

```diff
@@ -0,0 +1,229 @@
+  } else {
+    // Clear the tooltip content
+    el.value.textContent = null
+
+    // Hide the tooltip
+    el.value.style.removeProperty('--scalar-tooltip-offset')
+    el.value.style.setProperty('display', 'none')
+  }
+})
+
+/**
+ * Initialize the tooltip element
+ *
+ * If the tooltip is already initialized it will be ignored
+ */
+function initializeTooltipElement() {
+  if (el.value) {
+    // Tooltip already initialized
+    return
+  }
+
+  // See if the tooltip element already exists
+  // (Sometimes this happens with HMR)
+  const existingTooltipElement = document.getElementById(ELEMENT_ID)
+
+  if (existingTooltipElement) {
+    el.value = existingTooltipElement
+  } else {
+    // Create the tooltip element
+    el.value = document.createElement('div')
+    el.value.role = 'tooltip'
+    el.value.id = ELEMENT_ID
+    el.value.classList.add(ELEMENT_CLASS)
+    el.value.classList.add('scalar-app')
+    el.value.style.setProperty('display', 'none')
+    el.value.addEventListener('mouseleave', hideTooltip)
+    document.body.appendChild(el.value)
+  }
+}
+
+/**
+ * Show the tooltip after the delay if configured
+ */
+function showTooltipAfterDelay(_e: Event, opts: TooltipConfiguration) {
+  const delay = unref(opts.delay) ?? DEFAULT_DELAY
+
+  clearTimer()
+
+  // Show the tooltip after the delay
+  if (delay > 0) {
+    timer.value = setTimeout(() => showTooltip(_e, opts), delay)
+  } else {
+    showTooltip(_e, opts)
+  }
+}
+
+/**
+ * Show the tooltip
+ */
+function showTooltip(_e: Event, opts: TooltipConfiguration) {
+  clearTimer()
+
+  // Handle the escape key
+  document.addEventListener('keydown', handleEscape)
+
+  // Show the tooltip
+  config.value = opts
+}
+
+/**
+ * Hide the tooltip
+ *
+ * If the mouse is moving between the tooltip and the target we don't hide the tooltip
+ */
+function hideTooltip(_e: Event) {
+  if (!isMovingOffElements(_e)) {
+    // Don't hide the tooltip if the mouse is moving between the tooltip and the target
+    return
+  }
+
+  // Clear any existing timer
+  clearTimer()
+
+  // Hide the tooltip
+  config.value = undefined
+  document.removeEventListener('keydown', handleEscape)
+}
+
+/**
+ * Handle the escape key
+ *
+ * If the escape key is pressed we need to hide the tooltip
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/tooltip_role#keyboard_interactions
+ */
+function handleEscape(e: KeyboardEvent) {
+  if (e.key === 'Escape') {
+    hideTooltip(e)
+  }
+}
+
+/** Clears the current timer */
+function clearTimer() {
+  if (timer.value) {
+    clearTimeout(timer.value)
+    timer.value = undefined
+  }
+}
+
+/**
+ * Use the tooltip
+ *
+ * If the tooltip is not initialized it will be initialized
+ */
+export function useTooltip(opts: TooltipConfiguration) {
+  initializeTooltipElement()
+
+  const showThisTooltipAfterDelay = (e: Event) => showTooltipAfterDelay(e, opts)
+  const showThisTooltip = (e: Event) => showTooltip(e, opts)
+
+  watch(
```

Comment:
We should also see what the cost of this watcher is when its used in a tonne of components I know watchers used to be heavy (not sure what the current state is in latest vue version). 

We should do some benchmarking here by testing the sidebar on a big spec with and without this watcher

Reply:
would it be cheaper to just do a @hover on the element which needs the tooltip then show it? questions to ponder

Reply:
@hanspagel is the master benchmarker.

But we can just do a visual test of where the tooltip issue was found in the first place

Reply:
Sure, probably want to release for that though since it's in the other repo (and I changed a bunch of packages here)

Reply:
kk thats fine, iterative enhancements